sentence.go

```diff
@@ -97,9 +101,9 @@ func parsePrefix(s string) (string, string) {
 	return s[:2], s[2:]
 }
 
-// xor all the bytes in a string an return it
+// XORChecksum xor all the bytes in a string an return it
 // as an uppercase hex string
-func xorChecksum(s string) string {
+func XORChecksum(s string) string {
```

Comment:
I don't think this needs to be exported.

Reply:
If you look at my usage I am using it to create a command request.
https://github.com/arribada/smartLoraTracker/blob/3ae623f282f79d2b2f08045381a9e037d9ff6ab8/tools/gpsLora/gpsLora.go#L57

Reply:
@adrianmo your call

Reply:
Eh, thing function is 7 lines long. I think you're better off just copying it into your own code base. This package doesn't really provide much for encoding nmea messages.

Reply:
Why are you against exposing it?

Reply:
Hmm, I guess we do expose stuff like `ParseLatLong`, `ParseGPS`, etc ... so it's not completely unprecedented. Can you make a separate PR for it though?

Reply:
separate PR for such a small change?

Reply:
Feel free to link that PR to this issue.

Reply:
Sorry I didn't understand what you mean. 
My point was that I don't think a separate PR is needed for such a small non functional change.

Reply:
Aside from exporting the checksum function, I'm cool with merging this. So I want to close this PR, and continue discussion in a separate PR.

Reply:
Sure , I can open a PR, but I am just curious what is the reason you insist on having this in a separate PR for such a small non functional change?

Reply:
Because I'm not sure if I want to do it yet. I want to wait for @adrianmo for more input, but sometimes he doesn't show for a few days. In the meantime, I want to get the rest of your changes merged.

Reply:
I see, that makes sense. Thanks. PR to follow.

Reply:
updated.